lib/runcible/base.rb

```diff
@@ -43,15 +43,22 @@ def call(method, path, options = {})
       client_options[:verify_ssl] =  clone_config[:verify_ssl] unless clone_config[:verify_ssl].nil?
 
       if clone_config[:oauth]
-        headers = add_oauth_header(method, path, headers) if clone_config[:oauth]
+        headers = add_oauth_header(method, path, headers)
         headers['pulp-user'] = clone_config[:user]
-        client = RestClient::Resource.new(clone_config[:url], client_options)
+      elsif clone_config[:cert_auth]
+        if !clone_config[:cert_auth][:ssl_client_cert] || !clone_config[:cert_auth][:ssl_client_key]
+          fail Runcible::ConfigurationUndefinedError, "Missing necessary ssl information."
+        end
+        client_options[:ssl_client_cert] = clone_config[:cert_auth][:ssl_client_cert]
+        client_options[:ssl_client_key] = clone_config[:cert_auth][:ssl_client_key]
       else
         client_options[:user] =  clone_config[:user]
         client_options[:password] = config[:http_auth][:password]
-        client = RestClient::Resource.new(clone_config[:url], client_options)
       end
 
+      default_options[:ca_file] = config[:ca_cert_file] unless config[:ca_cert_file].nil?
```

Comment:
Is this change though? Look at where default_options is defined in the code this was taken from: https://github.com/Katello/runcible/pull/160/files#diff-85c6918fb17bafcb957a9367797d1c6fR168

`default_options` is nothing special, other than a local variable. What is passed to the client on the next line is what is important for configuring it properly, and in that case it is `client_options`